### PR TITLE
chore: adding isElementRendered

### PR DESCRIPTION
**Pull request by igorblyzn**

### CODE REVIEW

**Comment by igorblyzn** [comment]
@CrispusDH could you review please

File: API.md [hunk 1]
```diff
@@ -106,6 +107,14 @@ for sure amount of conditions can be increased endlessly
 
 check that element does not have attribute `disabled`
 
+#### isElementRendered
+- `page` Puppeteer's <Page> instance
+- `selector` <string> that represent CSS Selector
+- `timeout?` Optional. Default timeout is 10000ms
+- returns: `Promise<boolean>`
+
+check is element visible after timeout
```
Comment (CrispusDH) [inline API.md]: `check is element visible after timeout`
there is a difference between `visible` and `rendered` 
what exactly do you check?
as I see from code you check visibility
might be better to name that function differently
would leave it to you :D

File: src/predicates.ts [hunk 2]
```diff
@@ -3,50 +3,54 @@ import { pause } from './small';
 import { Predicate, waitFor } from './wait-for';
 import * as pFilter from 'p-filter';
 
-export const isElementFound = async (
+const checkVisibility = async (
```
Comment (CrispusDH) [inline src/predicates.ts]: let's put it inside `isElementVisible` as `const` variable with naming `isVisible` or smth

File: src/predicates.ts [hunk 3]
```diff
@@ -3,50 +3,54 @@ import { pause } from './small';
...
-    });
-    return true;
+    if (timeout) {
+      await waitFor(() => checkVisibility(page, selector), {
+        pollTime: 500,
+        timeout,
+      });
+      return true;
+    } else {
+      return await checkVisibility(page, selector);
```
Comment (CrispusDH) [inline src/predicates.ts]: `await` here is redundant

File: src/predicates.ts [hunk 4]
```diff
@@ -3,50 +3,54 @@ import { pause } from './small';
...
+    }
   } catch (error) {
-    error.message = `Element with ${selector} is not visible after ${timeout || 10000}ms timeout. Because of: ${error.message}`;
+    error.message = `Element with ${selector} is not visible. Because of: ${error.message}`;
```
Comment (CrispusDH) [inline src/predicates.ts]: what if user put `timeout`, do you think we should not add it to error message?

**Review by CrispusDH — CHANGES_REQUESTED** [review]

**Review by CrispusDH — APPROVED** [review]

**Pull request merged**